pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -210,70 +212,70 @@ type KubeControllerManagerConfiguration struct {
+	GenericComponent GenericComponentConfiguration `json:"genericComponent"`
 	// KubeCloudSharedConfiguration holds configuration for shared related features
 	// both in cloud controller manager and kube-controller manager.
-	KubeCloudShared KubeCloudSharedConfiguration
+	KubeCloudShared KubeCloudSharedConfiguration `json:"kubeCloudShared"`
 
 	// AttachDetachControllerConfiguration holds configuration for
 	// AttachDetachController related features.
-	AttachDetachController AttachDetachControllerConfiguration
+	AttachDetachController AttachDetachControllerConfiguration `json:"attachDetachController"`
 	// CSRSigningControllerConfiguration holds configuration for
 	// CSRSigningController related features.
-	CSRSigningController CSRSigningControllerConfiguration
+	CSRSigningController CSRSigningControllerConfiguration `json:"csrSigningController"`
 	// DaemonSetControllerConfiguration holds configuration for DaemonSetController
 	// related features.
-	DaemonSetController DaemonSetControllerConfiguration
+	DaemonSetController DaemonSetControllerConfiguration `json:"daemonSetController"`
 	// DeploymentControllerConfiguration holds configuration for
 	// DeploymentController related features.
-	DeploymentController DeploymentControllerConfiguration
+	DeploymentController DeploymentControllerConfiguration `json:"deploymentController"`
 	// DeprecatedControllerConfiguration holds configuration for some deprecated
 	// features.
-	DeprecatedController DeprecatedControllerConfiguration
+	DeprecatedController DeprecatedControllerConfiguration `json:"deprecatedController"`
 	// EndPointControllerConfiguration holds configuration for EndPointController
 	// related features.
-	EndPointController EndPointControllerConfiguration
+	EndPointController EndPointControllerConfiguration `json:"endPointController"`
 	// GarbageCollectorControllerConfiguration holds configuration for
 	// GarbageCollectorController related features.
-	GarbageCollectorController GarbageCollectorControllerConfiguration
+	GarbageCollectorController GarbageCollectorControllerConfiguration `json:"garbageCollectorController"`
 	// HPAControllerConfiguration holds configuration for HPAController related features.
-	HPAController HPAControllerConfiguration
+	HPAController HPAControllerConfiguration `json:"hpaController"`
 	// JobControllerConfiguration holds configuration for JobController related features.
-	JobController JobControllerConfiguration
+	JobController JobControllerConfiguration `json:"jobController"`
 	// NamespaceControllerConfiguration holds configuration for NamespaceController
 	// related features.
-	NamespaceController NamespaceControllerConfiguration
+	NamespaceController NamespaceControllerConfiguration `json:"namespaceController"`
 	// NodeIpamControllerConfiguration holds configuration for NodeIpamController
 	// related features.
-	NodeIpamController NodeIpamControllerConfiguration
+	NodeIpamController NodeIpamControllerConfiguration `json:"nodeIpamController"`
```

Comment:
`Ipam` or `IPAM`?

Reply:
IPAM indeed, but later

Reply:
same comment in https://github.com/kubernetes/kubernetes/pull/67090#discussion_r208331956 ... let's not add tags for serialization we intend to change